### ISSUE TITLE
max-classes-per-file: diagnostic anchored at the offending class declaration instead of Program start, breaking top-of-file `eslint-disable-next-line`

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25629]
## Summary

oxlint's native `max-classes-per-file` reports the diagnostic at the location of the _offending_ (second) `class` declaration, whereas ESLint reports it at the `Program` node (top of file). This off-by-many-lines location difference means an `// eslint-disable-next-line max-classes-per-file` placed on the line directly above the first statement — the canonical placement that ESLint honors — does NOT suppress oxlint's finding, and oxlint additionally flags the directive as an "Unused eslint-disable directive".

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0.

## Reproduction

`repro.ts`:

```ts
// eslint-disable-next-line max-classes-per-file
import {foo} from './foo'

const X = foo()

export class First {
	a = X
}

export class Second {
	b = X
}
```

`.oxlintrc.json`:

```json
{
	"plugins": [],
	"categories": {"correctness": "off"},
	"rules": {"max-classes-per-file": ["error", 1]}
}
```

Run:

```
oxlint -c .oxlintrc.json --report-unused-disable-directives repro.ts
```

## Expected

ESLint 7 and ESLint 9 both flag this file and report `max-classes-per-file` at the **start of the file (Program node, 1:1)**, so the `// eslint-disable-next-line` on the line directly preceding the first statement suppresses it and is counted as used — the run produces zero messages and exits 0. ESLint's implementation reports via `context.report({node, ...})` where `node` is the Program; anchoring the diagnostic there would restore parity.

## Actual

oxlint 1.78.0 reports:

```
repro.ts:10:8: error eslint(max-classes-per-file): File has too many classes (2). Maximum allowed is 1 help: Reduce the number of classes in this file
repro.ts:1:1: warning: Unused eslint-disable directive (no problems were reported).
```

The diagnostic is anchored on the second `class` declaration (line 10, the `class` keyword of `Second`) rather than the Program node, so the line-1 `disable-next-line` (which scopes to line 2 only) does not cover it — and the live, correctly placed directive is wrongly reported as unused. With the directive line removed, oxlint anchors at 9:8 (still the second class), confirming the anchor itself is the divergence.

## Cross-checked

| Tool | Version | Anchor (no directive) | Top-of-file `eslint-disable-next-line` |
| --- | --- | --- | --- |
| ESLint core rule | 7.23.0 (@typescript-eslint/parser 6.21.0) | 1:1 (Program) | suppresses, counted used — exit 0 |
| ESLint core rule | 9.39.4 flat config (typescript-eslint 8.62.1 parser) | 1:1 (Program) | suppresses, not reported unused — exit 0 |
| oxlint | 1.78.0 | 9:8 (second class) | does not suppress, reported unused — exit 1 |
